File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents;
    do
    {
        cents = get_int("Changed owed: \n");
    }
    while (cents < 0);

    // Calculate the number of quarters to give the customer
    int calculate_quarters(int cents);
    {
        int quarters = 0;

        while (cents >= 25)
        {
            quarters++;
            cents = cents - 25;
        }

        return quarters;
    }
// Calculate the number of dimes to give the customer
    int calculate_dimes(int cents);
    {
        int dimes = 0;

        while (cents >= 10)
        {
            dimes++;
            cents = cents - 10;
        }

        return dimes;
    }
// Calculate the number of nickels to give the customer
    int calculate_nickels(int cents);
    {
            int nickels = 0;

            while (cents >= 5)
            {
                nickels++;
                cents = cents - 5;
            }

            return nickels;
    }
// Calculate the number of pennies to give the customer
    int calculate_pennies(int cents);
    {
        int pennies = 0;

        while (cents >= 1)
        {
            pennies++;
            cents = cents -1;
        }

        return pennies;
    }
    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

File: Week_6/sentimental-cash/cash.py
# TODO
from cs50 import get_float


def main():
    # Prompting user for input
    owed = get_float("Change owed: ")
    # Keep prompting if non-negative integer is given
    if owed < 0:
        owed = get_float("Change owed: ")

    # Calculating functions call and rounding up owed variable to avoid floating point imprecision
    quarters = calc_quarters(round(owed, 2))
    owed = owed - (quarters * 0.25)

    dimes = calc_dimes(round(owed, 2))
    owed = owed - (dimes * 0.10)

    nickels = calc_nickels(round(owed, 2))
    owed = owed - (nickels * 0.05)

    pennies = calc_pennies(round(owed, 2))
    owed = owed - (pennies * 0.01)

    # Sum coins
    coins = quarters + dimes + nickels + pennies

    # Print total of coins
    print(f"{coins}")

# Calculate quarters


def calc_quarters(owed):
    quarters = 0
    while owed >= 0.25:
        quarters += 1
        owed = owed - 0.25
    return quarters

# Calculate dimes


def calc_dimes(owed):
    dimes = 0
    while owed >= 0.10:
        dimes += 1
        owed = owed - 0.10
    return dimes

# Calculate nickels


def calc_nickels(owed):
    nickels = 0
    while owed >= 0.05:
        nickels += 1
        owed = owed - 0.05
    return nickels

# Calculate pennies


def calc_pennies(owed):
    pennies = 0
    while owed >= 0.01:
        pennies += 1
        owed = owed - 0.01
    return pennies


main()

